player/command.c

```diff
@@ -3730,6 +3735,35 @@ static int mp_property_bindings(void *ctx, struct m_property *prop,
     return M_PROPERTY_NOT_IMPLEMENTED;
 }
 
+static int mp_property_mdata(void *ctx, struct m_property *prop,
+                                int action, void *arg)
+{
+    MPContext *mpctx = ctx;
+    mpv_node *node = &mpctx->command_ctx->mdata;
+
+    switch (action) {
+    case M_PROPERTY_GET_TYPE:
+        *(struct m_option *)arg = (struct m_option){.type = CONF_TYPE_NODE};
+        return M_PROPERTY_OK;
+    case M_PROPERTY_GET:
+    case M_PROPERTY_GET_NODE:
+        m_option_copy(&mdata_type, arg, node);
+        return M_PROPERTY_OK;
+    case M_PROPERTY_SET:
+    case M_PROPERTY_SET_NODE: {
+        m_option_copy(&mdata_type, node, arg);
```

Comment:
I think you need to free previous node, before copy. Else you will keep them all forever, until `command_ctx` is freed.

Reply:
It's already freed by `m_option_copy`, (which calls `opt->type->copy(opt, dst, src)`). 

```
    // Copy data between two locations. Deep copy if the data has pointers.
    // The implementation must free *dst if memory allocation is involved.
    /** \param opt The option to copy.
     *  \param dst Pointer to the destination memory.
     *  \param src Pointer to the source memory.
     */
    void (*copy)(const m_option_t *opt, void *dst, const void *src);
```